Sources/NetworkProtection/PacketTunnelProvider.swift

```diff
@@ -345,29 +349,40 @@ open class PacketTunnelProvider: NEPacketTunnelProvider {
         self.isConnectionTesterEnabled = vendorOptions?[NetworkProtectionOptionKey.connectionTesterEnabled] as? Bool ?? true
     }
 
-    private func loadKeyValidity(from options: [String: AnyObject]) {
-        guard let keyValidityString = options[NetworkProtectionOptionKey.keyValidity] as? String,
-              let keyValidity = TimeInterval(keyValidityString) else {
-            return
+    private func loadKeyValidity(from options: StartupOptions) {
+        switch options.keyValidity {
+        case .set(let validity):
+            setKeyValidity(validity)
+        case .useExisting:
+            break
+        case .reset:
+            setKeyValidity(nil)
```

Comment:
Now we have a basic mechanism to know whether we need to set the validity, use the existing one or reset it.